src/client/renderer/r_media.c

```diff
@@ -72,18 +74,27 @@ void R_DumpImage(const r_image_t *image, const char *output) {
 	glBindBuffer(GL_PIXEL_PACK_BUFFER, 0);
 	glBindBuffer(GL_PIXEL_UNPACK_BUFFER, 0);
 
-	R_BindDiffuseTexture(image->texnum);
+	// try GL_TEXTURE_2D first
+	target = GL_TEXTURE_2D;
+	R_BindUnitTexture(texunit_diffuse, image->texnum, target);
+
+	// and if GL_TEXTURE_2D failed, then it had to be GL_TEXTURE_2D_ARRAY
+	if (glGetError() == GL_INVALID_OPERATION) {
```

Comment:
Gross. There's no way to infer the correct target from the image itself? Seems like a miss.

Reply:
It should be possible by using it from image type, but there are some special virtual images that has most of it's fields filled with some kind of garbage.

Reply:
Ruh oh. Can you call any of them out? @Paril and I can probably get those fixed.

Reply:
I may do warning and skip with them

Reply:
None of them should be garbage, but, I think I recall what they're talking about. Atlasses are "virtual" images/they extend r_image_t, so that's probably what you're seeing there. Agree that target should be stored in r_image_t though.

Reply:
I'm looking for target by `layers` property which is missing for atlases. If that will be fixed then I might change this part to work correctly.

Reply:
Both r_atlas_t and r_atlas_image_t extend r_image_t, but I don't think they memset to zero. Either way you'll have to assign the target/layers in R_CompileAtlas and R_StitchAtlas, respectively. You can throw them under the "image.type = IT_ATLAS_" lines.